lib/logstash/inputs/redis.rb

```diff
@@ -268,15 +262,39 @@ def redis_runner
     begin
       @redis ||= connect
       yield
-    rescue ::Redis::BaseError => e
-      @logger.warn("Redis connection problem", :exception => e)
-      # Reset the redis variable to trigger reconnect
-      @redis = nil
-      Stud.stoppable_sleep(1) { stop? }
-      retry if !stop?
+    rescue => e
+      retry unless handle_error(e)
```

Comment:
A suggestion to rename `handle_error`
I can understand you want to retry when `handle_error` cannot handle, meaning return false
Looking at the whole function, we actually want to retry when it is retryable error, somehow meaning we can handle it, so it is a bit confusing at the first look.

may be we can do `retry handle_error(e)` or `retry retryable_error(e)` ?

Reply:
yeah, no super clear - also struggled a bit with naming here.
we're trying to re-use this method as much as possible.
so `handle_error` returns `true` if error is handled and should not be retried :wink: 

I guess you meant `retry if handle_error(e)`, 
that is actually what I started with but the inverse boolean condition [confused someone else](https://github.com/logstash-plugins/logstash-input-redis/pull/87/files/b19e1d9d54888bec9986530c623611fddab8bf64#r612638240) :rofl:

Reply:
I think `retry unless handle_error` is ok - another way you could possibly handle this is to just have the new method a no side effect logging method, like `log_error(e)`, which just deals with the formatting/logging keeping the actual retry logic (with the `ShutdownSignal` special case) local

Reply:
thanks, split the method into two. 
still needed some kind of `retry if ...` as the logic is at two places - did not want to have the same logic inline twice.
esp. since looking at the plugin the first time I manage to miss the "other" error handling place initially.

let's see what @kaisecheng thinks :crossed_fingers:

Reply:
This is much better. Splitting to two is great 👏 .  The naming and comment help a lot. Thanks for doing it.